### PR TITLE
Feature/business status

### PR DESCRIPTION
**Pull request by verafischer**

### CODE REVIEW

File: projects/angular-showcase/src/styles.scss [hunk 1]
```diff
@@ -141,3 +141,83 @@ nav.menu.ps .ps__rail-y {
     background-color: #fff;
   }
 }
+
+@mixin svgIconColors($fillColor, $strokeColor) {
```
Comment (verafischer) [inline projects/angular-showcase/src/styles.scss]: Where else to put these style definitions?

**Comment by kyubisation** [comment]
@verafischer Thank you for the contribution 😃 
I'll try to review this as soon as possible.

File: projects/sbb-esta/angular-business/status/src/status/status.component.html [hunk 2]
```diff
@@ -0,0 +1,20 @@
+<ng-template #content><ng-content></ng-content></ng-template>
```
Comment (kyubisation) [inline projects/sbb-esta/angular-business/status/src/status/status.component.html]: The design specification (currently) specifies the states valid, warning, invalid (and consumer specific variants). I would like to provide these by default without having the consumer provide the icons themselves. The consumer should only provide the icons for consumer specific variants.

File: projects/sbb-esta/angular-business/status/src/status/status.component.scss [hunk 3]
```diff
@@ -0,0 +1,25 @@
+@import '../../../../angular-core/styles/common/variables';
+@import '../../../../angular-core/styles/common/colors';
+
+:host {
+  display: flex;
```
Comment (kyubisation) [inline projects/sbb-esta/angular-business/status/src/status/status.component.scss]: I would expect the status to be an inline element, so in this case flex-inline would be preferred.

File: projects/sbb-esta/angular-business/status/src/status/status.component.scss [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  flex-direction: row;
+
+  font-family: $fontSbbRoman;
+  font-size: 0.9375rem;
```
Comment (kyubisation) [inline projects/sbb-esta/angular-business/status/src/status/status.component.scss]: Feel free to use the pxToRem function. This way we can use the values provided by the design specification (which are usually in pixel), while also providing rem values for consumers.

File: projects/sbb-esta/angular-business/status/src/status/status.component.spec.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
+import { NO_ERRORS_SCHEMA } from '@angular/core';
```
Comment (kyubisation) [inline projects/sbb-esta/angular-business/status/src/status/status.component.spec.ts]: Why is this required?

File: projects/sbb-esta/angular-business/status/src/status/status.component.ts [hunk 6]
```diff
@@ -0,0 +1,31 @@
+import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
+
+@Component({
+  selector: 'sbb-status',
+  templateUrl: './status.component.html',
+  styleUrls: ['./status.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class StatusComponent {
+  /**
```
Comment (kyubisation) [inline projects/sbb-esta/angular-business/status/src/status/status.component.ts]: Type is missing: https://github.com/sbb-design-systems/sbb-angular/issues/86

File: projects/sbb-esta/angular-business/status/src/status/status.component.ts [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+   * Set the status message shown on the right side.
+   */
+  @Input() message: string;
+
+  /**
+   * Set the status tooltip text. If the text is set then the tooltip appears on hover.
+   */
+  @Input() tooltipText: string;
```
Comment (kyubisation) [inline projects/sbb-esta/angular-business/status/src/status/status.component.ts]: As mentioned previously, tooltip is not part of this spec

File: projects/angular-showcase/src/app/business/business-examples/status-showcase/status-showcase.component.html [hunk 8]
```diff
@@ -1,3 +1,17 @@
+<h3>Status with type</h3>
+<p>
+  <sbb-status [type]="validType" [message]="'valid type'"></sbb-status>
```
Comment (kyubisation) [inline projects/angular-showcase/src/app/business/business-examples/status-showcase/status-showcase.component.html]: Here I would prefer assigning the type as an attribute, to show the simple usage
e.g. `<sbb-status type="valid" ...`

File: projects/angular-showcase/src/app/business/business-examples/status-showcase/status-showcase.component.html [hunk 9]
```diff
@@ -54,18 +66,9 @@ <h3>Status Extended used in table</h3>
   <tbody>
     <tr *ngFor="let row of rows">
```
Comment (kyubisation) [inline projects/angular-showcase/src/app/business/business-examples/status-showcase/status-showcase.component.html]: Having a dynamic table for an example with four entries is a bit overkill.
I'd prefer to just have the four rows.
This is also better for consumer who would like to see the source of the example.

File: projects/sbb-esta/angular-business/status/src/status.module.ts [hunk 10]
```diff
@@ -12,10 +17,13 @@ import { StatusComponent } from './status/status.component';
 @NgModule({
   imports: [
     CommonModule,
-    IconCollectionModule,
     OverlayModule,
-    BrowserAnimationsModule,
-    BrowserModule
+    BrowserModule,
```
Comment (kyubisation) [inline projects/sbb-esta/angular-business/status/src/status.module.ts]: Remove the BrowserModule. A component module should not contain it.

File: projects/sbb-esta/angular-business/status/src/status/status-type.enum.ts [hunk 11]
```diff
@@ -0,0 +1,5 @@
+export enum StatusType {
```
Comment (kyubisation) [inline projects/sbb-esta/angular-business/status/src/status/status-type.enum.ts]: I don't think we have something like this for other modules. Is there a specific reason why you added this?

File: projects/sbb-esta/angular-business/status/src/status/status.component.ts [hunk 12]
```diff
@@ -1,17 +1,28 @@
...
+import { StatusType } from './status-type.enum';
+
 @Component({
   selector: 'sbb-status',
   templateUrl: './status.component.html',
   styleUrls: ['./status.component.scss'],
   changeDetection: ChangeDetectionStrategy.OnPush
 })
 export class StatusComponent {
+  public readonly validType = StatusType.VALID;
```
Comment (kyubisation) [inline projects/sbb-esta/angular-business/status/src/status/status.component.ts]: These properties are not necessary.

File: projects/sbb-esta/angular-business/status/src/status/status.component.ts [hunk 13]
```diff
@@ -1,17 +1,28 @@
...
+
   /**
    * Set the status message shown on the right side.
    */
   @Input() message: string;
 
+  /**
+   * If the type is set the icon of this status is shown
+   */
+  @Input() type: StatusType;
```
Comment (kyubisation) [inline projects/sbb-esta/angular-business/status/src/status/status.component.ts]: Use string literals as type for inputs: `'valid' | 'warning' | 'invalid'`

File: projects/sbb-esta/angular-business/status/status.md [hunk 14]
```diff
@@ -2,14 +2,35 @@ The status shows the status of an object.
...
+### Status with type
 
-Set an `sbb-icon` tag as content inside the `<sbb-status>`.
+You can set a predefined type `[type]`.
+Predefined types are defined in the StatusType enum.
+At the time there are `'valid'`, `'invalid'` and `'warning'` predefined types.
 
 ```html
-<sbb-status>
+<sbb-status [type]="'valid'" [message]="'valid type'"></sbb-status>
```
Comment (kyubisation) [inline projects/sbb-esta/angular-business/status/status.md]: Use the attribute variant as example here `<sbb-status type="valid" ...`

File: projects/sbb-esta/angular-business/styles.scss [hunk 15]
```diff
@@ -1,3 +1,21 @@
 $sbbBusiness: true;
 
 @import '../angular-core/styles/_main.scss';
+
```
Comment (kyubisation) [inline projects/sbb-esta/angular-business/styles.scss]: This should be moved to the status.component.scss, either with ::ng-deep or with ViewEncapsulation.None

**Comment by efux** [comment]
Hello @kyubisation and @verafischer. We are planning on using the Tooltip component in our projects. In our use case we would like to use it without an icon and without the close icon when hovering over the element. I guess this will be fixed by this PR. Is there still some progress for this issue/PR?

**Comment by kyubisation** [comment]
Migrated changes to #448
